src/Type/Type.php

```diff
@@ -48,6 +48,10 @@ public function getIterableKeyType(): Type;
 
 	public function getIterableValueType(): Type;
 
+	public function isOffsetAccesible(): TrinaryLogic;
+
+	public function getOffsetValueType(): Type;
```

Comment:
This is incompatible with constant types as they will need `getOffsetValueType(Type $offsetType)` signature. You can keep the current and we will change it in constant type implementation or may be make it future proof by adding the extra parameter you currently do not need.